website/public/js/controllers/groupsCtrl.js

```diff
@@ -16,6 +16,15 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
     }
 
     $scope.isMemberOfGroup = function(userid, group){
+
+      // If the group is a guild, just check for an intersection with the
+      // current user's guilds, rather than checking the members of the group.
+      if(group.type === 'guild') {
+        if (_.detect(Groups.myGuilds(), function(g) { return g._id === group._id })) {
+          return true;
```

Comment:
Yeah, that makes sense. I was coding defensively - as I wasn't sure whether `myGuilds` is ever truncated, so I figured I'd just let it fall through. But looking at the code further, that's not the case :)